examples/data_assimilation.jl

```diff
@@ -0,0 +1,173 @@
+# # Calibrating a biogeochemical model with `EnsembleKalmanProcesses`
+#
+# In this example we calibrate some of the parameters for the [NPZD](@ref NPZD) model
+# in a simple box model setup using a data assimilaiton package called [EnsembleKalmanProcesses](https://github.com/CliMA/EnsembleKalmanProcesses.jl).
+# First we setup the model and generate synthetic data with "true" parameters. We then
+# define priors and setup an EKP to solve.
+#
+# While this is a very simple situation it illustraites the ease of integration with 
+# data assimilation tools. Examples given in the EnsembleKalmanProcesses docs illustraite
+# how the package can be used to solve more complex forward models.
+
+# ## Install dependencies
+# First we ensure we have the required dependencies installed
+# ```julia
+# using Pkg
+# pkg "add OceanBioME, Oceananigans, CairoMakie, EnsembleKalmanProcesses, Distributions"
+# ```
+
+using OceanBioME, EnsembleKalmanProcesses, JLD2, CairoMakie, Oceananigans.Units
+using LinearAlgebra, Random
+
+using Distributions
+
+using EnsembleKalmanProcesses
+using EnsembleKalmanProcesses.ParameterDistributions
+
+const year = years = 365day
+
+rng_seed = 41
+rng = Random.MersenneTwister(rng_seed)
+
+# Setup the forward model
+
+@inline PAR⁰(t) = 60 * (1 - cos((t + 15days) * 2π / year)) * (1 / (1 + 0.2 * exp(-((mod(t, year) - 200days) / 50days)^2))) + 2
+
+@inline PAR(t) = PAR⁰(t) * exp(- 0.2 * 10)
+
+function model(initial_photosynthetic_slope, 
+               base_maximum_growth, 
+               nutrient_half_saturation, 
+               phyto_base_mortality_rate, 
+               j)
+
+    biogeochemistry = NutrientPhytoplanktonZooplanktonDetritus(; grid = BoxModelGrid(),
+                                                                 initial_photosynthetic_slope, 
+                                                                 base_maximum_growth, 
+                                                                 nutrient_half_saturation, 
+                                                                 phyto_base_mortality_rate)
+
+    model = BoxModel(; biogeochemistry, forcing = (; PAR))
+
+    model.Δt = 20minutes
+    model.stop_time = 5years
+
+    set!(model, N = 10.0, P = 0.1, Z = 0.01)
+
+    run!(model, save_interval = 24, feedback_interval = Inf, save = SaveBoxModel("box_calibration_$j.jld2"))
+
+    file = jldopen("box_calibration_$j.jld2")
+    times = parse.(Float64, keys(file["values"]))
+
+    times = times[length(times)-1092:end]
+
+    P = zeros(1093)
+
+    for (idx, t_idx) in enumerate(length(times)-1092:length(times))
+        P[idx] = file["values/$(times[t_idx])"].P
```

Comment:
Why does the "time" have a field `.P`?

Reply:
P is a property of a tuple of values saved as values/timestep index like how in FieldTimeSeries we have `timeseries` which then has timesteps and in each `timeseries/time step index` you get each tracer.

Reply:
Why not use the file address `timeseries/iteration/name` here?

Reply:
Good question, I'll change that for consitency too